### PR TITLE
release_gen_assembly.py option to output to file

### PR DESCRIPTION
**Pull request by ashwindasr**
Option to write the output of gen-assembly doozer command to a file. Useful for https://github.com/openshift/aos-cd-jobs/pull/3140 as its better to read from a file rather than the output to create the PR.

### CODE REVIEW

File: doozerlib/cli/release_gen_assembly.py [hunk 1]
```diff
@@ -46,9 +46,10 @@ def releases_gen_assembly(ctx, name):
 @click.option("--suggestions-url", metavar='SUGGESTIONS_URL', required=False,
               default="https://raw.githubusercontent.com/openshift/cincinnati-graph-data/master/build-suggestions/",
               help="When using --auto-previous, set custom suggestions URL, load from {major}-{minor}-{arch}.yaml")
+@click.option('--output-file', required=False, help='Specify a file path to write the doozer output to')
```
Comment (thegreyd) [inline doozerlib/cli/release_gen_assembly.py:49]: "to write the generated assembly definition to" instead of doozer output

Comment (thegreyd) [reply doozerlib/cli/release_gen_assembly.py:49]: Also we can add a shortflag `-o`

Comment (ashwindasr) [reply doozerlib/cli/release_gen_assembly.py]: Done!

File: doozerlib/cli/release_gen_assembly.py [hunk 2]
```diff
@@ -398,3 +399,7 @@ def exit_with_error(msg):
     }
 
     print(yaml.dump(assembly_def))
+
+    if output_file:
+        with open(output_file, 'w') as file:
```
Comment (thegreyd) [inline doozerlib/cli/release_gen_assembly.py:404]:
```suggestion
        stream = file(output_file, 'w')
        yaml.dump(data, stream)
```

Comment (thegreyd) [reply doozerlib/cli/release_gen_assembly.py:404]: Essentially the same but more concise

Comment (ashwindasr) [reply doozerlib/cli/release_gen_assembly.py:404]: Done!

**Pull request merged**